### ISSUE TITLE
Standardise naming and documentation of Exceptions and Errors 

### ISSUE THREAD

**Issue by ravernkoh** [crystal-lang/crystal#5559]
# Inconsistent naming
Looking through the [Standard Library API](https://crystal-lang.org/api/0.24.1/), I noticed some inconsistent naming between `Exception`s and `Error`s. After asking [this question](https://stackoverflow.com/questions/48142967/exceptions-errors-in-crystal) on StackOverflow, it appears `Error`s should be used for specialized subclasses of `Exceptions`. However, this isn't the case for the `JSON` module (as pointed out in the question), where `JSON::ParseException > JSON::Error > Exception`.

# Lack of documentation
I understand that this is a new language and I can't expect documentation to be extensive, but I feel that there should at least be some mention of the possible `Exception`s/`Error`s a method could possibly raise. For example, with the `File.new` method, I need to have some prior knowledge in order to know that opening files usually raise exceptions on err. Also, in order to check what can be raised, I have to either look at source code, or try to make the method crash for myself.

# Questions/Suggestions
* At this stage, is it possible to rename `Error`s and `Exceptions`s to be a little more consistent? And also document the conventions somewhere?
* Standardize a way to indicate possible `Error`s and `Exception`s a method can raise?

P.S. I would gladly make a PR and work on this issue if other people feel the same way.

**Comment by shevegen** [comment]
> I understand that this is a new language and I can't expect documentation
> to be extensive

Ruby in many ways has a worse documentation. ;)

Crystal at the least has an intro-tutorial that is quite ok IMO.

I think in many ways, documentation is not only extremely useful but
als hugely important, in particular for newcomers (it's not as important
for people who may have been working with a language for years;
I collected my own documentation for ruby, for example).

**Comment by jwaldrip** [comment]
I would say a much better conversation is that in the form of a PR. The
beauty of this language is the ability and ease of having anyone contribute.

**Comment by ravernkoh** [comment]
> At this stage, is it possible to rename `Error`s and `Exception`s to be a little more consistent? And also document the conventions somewhere?

Yeah, that's why I asked this question. Renaming the class would result in a breaking change.

**Comment by RX14** [comment]
The general consensus is to name everything `FooError` (assuming it's actually an error) but keep the top-level class called `Exception`. Not all exceptions have to be errors.

**Comment by ravernkoh** [comment]
At this point, is it okay to make a PR consisting of breaking changes to the Standard Library API (Like renaming the `JSON::ParseException` class)? Is there any label or etc. that I need to mark the breaking change?

**Comment by RX14** [comment]
@ravernkoh just make the PR :)
